// formilk/dist/chunk-ZLHKWKNP.js
import {
  custom_elements_default
} from "./chunk-HM7LLTML.js";

// src/button/button-active.tsx
import { jsx as _jsx } from "atomico/jsx-runtime";
import { c, css, useState } from "atomico";
import { getCoordinates } from "@atomico/hooks/use-click-coordinates";
function buttonActive() {
  const [state, setState] = useState();
  return /* @__PURE__ */ _jsx("host", {
    shadowDom: true,
    setEvent: (event) => {
      const { offset } = getCoordinates(event);
      setState((state2) => state2?.pending ? state2 : { offset, pending: true });
    }
  }, /* @__PURE__ */ _jsx("div", {
    style: state?.offset ? `--x:${state?.offset?.x}px; --y:${state?.offset?.y}px` : "",
    class: state?.pending ? "show" : "",
    onanimationend: () => setState({})
  }));
}
buttonActive.styles = css`:host {
        width: 100%;
        height: 100%;
        position: absolute;
        top: 0px;
        left: 0px;
        display: block;
        overflow: hidden;
        --background: rgba(0, 0, 0, 0.05);
    }
    div {
        width: 10px;
        height: 10px;
        position: absolute;
        top: calc(var(--y) - 5px);
        left: calc(var(--x) - 5px);
        opacity: 0;
        background: var(--background);
        border-radius: 100%;
    }
    .show {
        transform: scale(2);
        opacity: 1;
        animation: scale 0.5s ease 1;
    }
    @keyframes scale {
        0% {
            opacity: 0;
            transform: scale(0);
        }
        50% {
            opacity: 1;
        }
        100% {
            transform: scale(10);
            opacity: 0;
        }
    }`;
var ButtonActive = c(buttonActive);
custom_elements_default.define("button-active", ButtonActive);

export {
  ButtonActive
};
