// formilk/dist/chunk-QL73EMIW.js
import {
  custom_elements_default
} from "./chunk-HM7LLTML.js";

// src/input-file/input-file-preview.tsx
import { jsx as _jsx } from "atomico/jsx-runtime";
import { c, css } from "atomico";
function inputFilePreview({ file }) {
  const src = file && URL.createObjectURL(file);
  return /* @__PURE__ */ _jsx("host", {
    shadowDom: true
  }, file && (/^video/.test(file.type) ? /* @__PURE__ */ _jsx("video", {
    class: "media",
    src,
    controls: true
  }) : /^image/.test(file.type) ? /* @__PURE__ */ _jsx("img", {
    class: "media",
    src
  }) : /^audio/.test(file.type) ? /* @__PURE__ */ _jsx("audio", {
    class: "media",
    src,
    controls: true
  }) : null));
}
inputFilePreview.props = {
  slot: {
    type: String,
    reflect: true,
    value: "preview"
  },
  file: File
};
inputFilePreview.styles = css`:host {
        display: flex;
        align-items: center;
    }
    .media {
        max-width: 100%;
        display: block;
        margin: auto;
    }`;
var InputFilePreview = c(inputFilePreview);
custom_elements_default.define("input-file-preview", InputFilePreview);

export {
  InputFilePreview
};
