// formilk/dist/chunk-AYN55A6G.js
import {
  InputCheckbox
} from "./chunk-ZVYBBMSC.js";
import {
  useCheckbox
} from "./chunk-2NV7RVDQ.js";
import {
  custom_elements_default
} from "./chunk-HM7LLTML.js";

// src/input-switch/input-switch.tsx
import { jsx as _jsx } from "atomico/jsx-runtime";
import { c, css, useHost } from "atomico";
import { useDisabled } from "@atomico/hooks/use-disabled";
import { useReflectEvent } from "@atomico/hooks/use-reflect-event";
function inputSwitch() {
  const host = useHost();
  const refInput = useCheckbox("checkbox");
  const disabled = useDisabled();
  useReflectEvent(host, refInput, "click");
  return /* @__PURE__ */ _jsx("host", {
    shadowDom: true
  }, /* @__PURE__ */ _jsx("button", {
    class: "checkbox",
    disabled
  }, /* @__PURE__ */ _jsx("div", {
    class: "checkbox-state"
  }, /* @__PURE__ */ _jsx("slot", {
    name: "icon"
  }))));
}
inputSwitch.styles = css`:host {
        --border-radius: 100px;
        ---state-opacity: 0.25;
        ---width: calc((var(--size-s) * 2) + (var(--size-m) - var(--size-s)));
        ---state-size: var(--size-s);
        ---state-radius: 100%;
        ---state-transform: translateX(-50%);
    }
    :host([checked]) {
        ---state-transform: translateX(50%);
    }
    .checkbox {
        width: var(---width);
    }`;
var InputSwitch = c(inputSwitch, InputCheckbox);
custom_elements_default.define("input-switch", InputSwitch);

export {
  InputSwitch
};
