// formilk/dist/chunk-BV7KU7BG.js
import {
  custom_elements_default
} from "./chunk-HM7LLTML.js";

// src/input-select/input-select-option.tsx
import { jsx as _jsx } from "atomico/jsx-runtime";
import { useSlot } from "@atomico/hooks/use-slot";
import { c, useRef } from "atomico";
function selectOption() {
  const refSlotOption = useRef();
  const slotOption = useSlot(refSlotOption);
  return /* @__PURE__ */ _jsx("host", {
    shadowDom: true,
    options: slotOption
  }, /* @__PURE__ */ _jsx("slot", {
    ref: refSlotOption,
    name: "option"
  }));
}
var eventOptionChange = { type: "OptionChange", bubbles: true };
selectOption.props = {
  slot: {
    type: String,
    reflect: true,
    value: "option"
  },
  value: {
    type: String,
    event: eventOptionChange
  },
  label: {
    type: String,
    event: eventOptionChange
  },
  options: {
    type: Array,
    event: eventOptionChange
  },
  selected: Boolean
};
var InputSelectOption = c(selectOption);
custom_elements_default.define("input-select-option", InputSelectOption);

export {
  InputSelectOption
};
