// formilk/dist/chunk-ZVYBBMSC.js
import {
  useCheckbox
} from "./chunk-2NV7RVDQ.js";
import {
  Icon
} from "./chunk-LXG56IO5.js";
import {
  InputGenericProps
} from "./chunk-YDOX3GBQ.js";
import {
  tokens_default
} from "./chunk-PXPI3F43.js";
import {
  custom_elements_default
} from "./chunk-HM7LLTML.js";

// src/input-checkbox/input-checkbox.tsx
import { jsx as _jsx } from "atomico/jsx-runtime";
import { c, css, useHost } from "atomico";
import { useDisabled } from "@atomico/hooks/use-disabled";
import { useReflectEvent } from "@atomico/hooks/use-reflect-event";
function checkbox({ tabIndex }) {
  const host = useHost();
  const refInput = useCheckbox("checkbox");
  const disabled = useDisabled();
  useReflectEvent(host, refInput, "click");
  return /* @__PURE__ */ _jsx("host", {
    shadowDom: true
  }, /* @__PURE__ */ _jsx("button", {
    class: "checkbox",
    disabled,
    tabIndex: disabled ? -1 : tabIndex
  }, /* @__PURE__ */ _jsx("div", {
    class: "checkbox-state"
  }, /* @__PURE__ */ _jsx(Icon, {
    type: "check"
  }))));
}
checkbox.props = {
  ...InputGenericProps,
  value: {
    type: null,
    value: "on"
  },
  tabIndex: { type: Number },
  checked: { type: Boolean, reflect: true },
  id: String
};
checkbox.styles = [
  tokens_default,
  css`:host {
            --color-input-30: var(--color-input-10);
            ---state-opacity: 0;
            ---state-size: 100%;
            cursor: pointer;
        }

        :host([disabled]) {
            opacity: var(--opacity-disabled);
            pointer-events: none;
        }

        :host([checked]) {
            ---state-opacity: 1;
        }

        .checkbox {
            width: var(--size-m);
            height: var(--size-m);
            padding: 0;
            background: var(--color-input-60);
            border-radius: var(--border-radius);
            border: var(--input-border-width) var(--input-border-style)
                var(--color-input-30);
            overflow: hidden;
            cursor: unset;
            font: unset;
            display: flex;
        }

        .checkbox-state {
            width: var(---state-size);
            height: var(---state-size);
            background: var(--color-input-10);
            opacity: var(---state-opacity);
            margin: auto;
            border-radius: var(---state-radius);
            transform: var(---state-transform);
            display: flex;
            place-content: center;
        }

        .checkbox-state * {
            color: var(--color-input-60);
        }

        .checkbox,
        .checkbox-state {
            transition: var(--transition-action);
        }`
];
var InputCheckbox = c(checkbox);
custom_elements_default.define("input-checkbox", InputCheckbox);

export {
  InputCheckbox
};
