// formilk/dist/chunk-JSWF7MAP.js
import {
  Icon
} from "./chunk-LXG56IO5.js";
import {
  tokens_default
} from "./chunk-PXPI3F43.js";
import {
  custom_elements_default
} from "./chunk-HM7LLTML.js";

// src/avatar/avatar.tsx
import { jsx as _jsx } from "atomico/jsx-runtime";
import { c, css } from "atomico";
function avatar({ src, size, transform }) {
  return /* @__PURE__ */ _jsx("host", {
    shadowDom: true
  }, /* @__PURE__ */ _jsx("button", {
    class: "avatar-mask",
    part: "avatar-mask"
  }, /* @__PURE__ */ _jsx("div", {
    class: "avatar-inner",
    part: "avatar-inner"
  }, /* @__PURE__ */ _jsx("slot", null, src ? /* @__PURE__ */ _jsx("img", {
    class: "avatar-img",
    src
  }) : /* @__PURE__ */ _jsx(Icon, {
    type: "avatar"
  })))), /* @__PURE__ */ _jsx("style", null, !!size && `:host{--size:var(--size-${size});}`, !!transform && `:host{--transform:${transform};}`));
}
avatar.props = {
  src: String,
  size: {
    type: String,
    reflect: true
  },
  transform: {
    type: String,
    reflect: true
  }
};
avatar.styles = [
  tokens_default,
  css`.avatar-mask {
            width: var(--size-xl);
            height: var(--size-xl);
            overflow: hidden;
            border-radius: var(--border-radius);
            background: var(--color-layer-60);
            border: none;
            padding: 0px;
            cursor: unset;
            margin: auto;
        }
        .avatar-inner {
            width: 100%;
            height: 100%;
            transform: var(--transform);
            place-content: center;
            display: flex;
        }
        ::slotted(*),
        .avatar-img {
            width: 100%;
            height: 100%;
            object-fit: cover;
            display: block;
        }`
];
var Avatar = c(avatar);
custom_elements_default.define("avatar", Avatar);

export {
  Avatar
};
